src/main/java/org/jabref/logic/pdf/search/PdfIndexer.java

```diff
@@ -45,16 +48,33 @@ public class PdfIndexer {
     private static final Logger LOGGER = LoggerFactory.getLogger(PdfIndexer.class);
 
     @VisibleForTesting
+    @Nullable // null might happen if lock is held by another JabRef instance
     IndexWriter indexWriter;
 
     private final BibDatabaseContext databaseContext;
+
     private final FilePreferences filePreferences;
+
+    @Nullable
     private final Directory indexDirectory;
+
     private IndexReader reader;
 
     private PdfIndexer(BibDatabaseContext databaseContext, Directory indexDirectory, FilePreferences filePreferences) {
         this.databaseContext = databaseContext;
-        this.indexDirectory = indexDirectory;
+        if (indexDirectory == null) {
```

Comment:
I checked the code. Especially `org.jabref.model.database.BibDatabaseContext#getFulltextIndexPath`. That never returns null.

Proposal: Ship that code and "check" if at some persons report that output. (Line 67). Then, we can track down the issue.